azurestack/resource_arm_route.go

```diff
@@ -0,0 +1,178 @@
+
+	azureStackLockByName(rtName, routeTableResourceName)
+	defer azureStackUnlockByName(rtName, routeTableResourceName)
+
+	properties := network.RoutePropertiesFormat{
+		AddressPrefix: &addressPrefix,
+		NextHopType:   network.RouteNextHopType(nextHopType),
+	}
+
+	if v, ok := d.GetOk("next_hop_in_ip_address"); ok {
+		nextHopInIpAddress := v.(string)
+		properties.NextHopIPAddress = &nextHopInIpAddress
+	}
+
+	route := network.Route{
+		Name: &name,
+		RoutePropertiesFormat: &properties,
+	}
+
+	future, err := client.CreateOrUpdate(ctx, resGroup, rtName, name, route)
+	if err != nil {
+		return fmt.Errorf("Error Creating/Updating Route %q (Route Table %q / Resource Group %q): %+v", name, rtName, resGroup, err)
+	}
+
+	err = future.WaitForCompletionRef(ctx, client.Client)
+	if err != nil {
+		return fmt.Errorf("Error waiting for completion for Route %q (Route Table %q / Resource Group %q): %+v", name, rtName, resGroup, err)
+	}
+
+	read, err := client.Get(ctx, resGroup, rtName, name)
+	if err != nil {
+		return err
+	}
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read Route %q/%q (resource group %q) ID", rtName, name, resGroup)
+	}
+	d.SetId(*read.ID)
+
+	return resourceArmRouteRead(d, meta)
+}
+
+func resourceArmRouteRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).routesClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resGroup := id.ResourceGroup
+	rtName := id.Path["routeTables"]
+	routeName := id.Path["routes"]
+
+	resp, err := client.Get(ctx, resGroup, rtName, routeName)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error making Read request on Azure Route %q: %+v", routeName, err)
+	}
+
+	d.Set("name", routeName)
+	d.Set("resource_group_name", resGroup)
+	d.Set("route_table_name", rtName)
+
+	if props := resp.RoutePropertiesFormat; props != nil {
+		d.Set("address_prefix", props.AddressPrefix)
+		d.Set("next_hop_type", string(props.NextHopType))
+
+		if ip := props.NextHopIPAddress; ip != nil {
+			d.Set("next_hop_in_ip_address", props.NextHopIPAddress)
+		}
+	}
+
+	return nil
+}
+
+func resourceArmRouteDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).routesClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resGroup := id.ResourceGroup
+	rtName := id.Path["routeTables"]
+	routeName := id.Path["routes"]
+
+	azureStackLockByName(rtName, routeTableResourceName)
+	defer azureStackUnlockByName(rtName, routeTableResourceName)
+
+	future, err := client.Delete(ctx, resGroup, rtName, routeName)
+	if err != nil {
+		return fmt.Errorf("Error deleting Route %q (Route Table %q / Resource Group %q): %+v", routeName, rtName, resGroup, err)
+	}
+
+	err = future.WaitForCompletionRef(ctx, client.Client)
+	if err != nil {
```

Comment:
this also could be: `if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {`